### PR TITLE
Compatibility with Rails 4.2

### PR DESCRIPTION
**Pull request by grosser**
@pschambacher @zendesk/zendesk-rails-upgraders 

first few obvious patches ... needs more work ...

### Risks
 - None

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@ gemfile:
   - gemfiles/activerecord_3.2.gemfile
   - gemfiles/activerecord_4.0.gemfile
   - gemfiles/activerecord_4.1.gemfile
+  - gemfiles/activerecord_4.2.gemfile
```
Comment (pschambacher) [inline .travis.yml]: Would be nice to apply usual naming convention

File: kasket.gemspec [hunk 2]
```diff
@@ -10,7 +10,7 @@ Gem::Specification.new do |s|
   s.description = "puts a cap on your queries"
   s.license     = "Apache License Version 2.0"
 
-  s.add_runtime_dependency("activerecord", ">= 3.2", "< 4.2")
+  s.add_runtime_dependency("activerecord", ">= 3.2", "< 4.3")
```
Comment (pschambacher) [inline kasket.gemspec]: 5.0 should do. Let's remove alias_method_chain if any while we're here

File: .travis.yml (hunk 1 above)
Comment (pschambacher) [inline .travis.yml]: Ruby 1.9.3 can probably die as well so we can use prepend

Comment (grosser) [inline .travis.yml]: yeah, trying to keep the diff low until it's green :D

**Comment by pschambacher** [comment]
There it's green now :smile: 
/cc @fneves @osheroff @craig-day

File: lib/kasket/configuration_mixin.rb [hunk 3]
```diff
@@ -37,7 +37,8 @@ def kasket_key_for(attribute_value_pairs)
           column = columns_hash[attribute.to_s]
           value = nil if value.blank?
           quoted_value = if column
-            connection.quote(column.type_cast(value), column).downcase
+            casted_value = column.respond_to?(:type_cast_for_database) ? column.cast_type.send(:type_cast, value) : column.type_cast(value)
```
Comment (grosser) [inline lib/kasket/configuration_mixin.rb]: maybe an `if column.respond_to?(:type_cast_for_database) # Rails 4.2+` to show why we are doing this

**Comment by grosser** [comment]
:+1: :heart:

**Comment by fneves** [comment]
:heart:

**Comment by pschambacher** [comment]
@staugaard @grosser I can't release this gem :sweat_smile:

**Pull request merged**